templates/rustdoc/body.html

```diff
@@ -29,4 +29,16 @@
       window.addEventListener("scroll", maybeFixupViewPortPosition, {"once": true});
     }
   }
+
+  window.addEventListener('message', function (ev) {
+    if (ev.data && ev.data.storage && ev.data.storage.key === 'rustdoc-theme') {
+      applyTheme(ev.data.storage.value);
+    }
+  });
 </script>
+<!--
+    Only other windows get notified when we change local storage, so we have an
+    invisible iframe that sends us a message when local storage changes so we
+    can detect rustdoc changing the theme
```

Comment:
> Only other windows get notified when we change local storage

Who came up with that idea??? 🤦 

Well, if that's the case this looks fine.

Reply:
Yeah, and I was _very_ surprised to find out that neither dedicated or shared workers get the events either, even though they should be treated as a separate context, requiring going all the way to loading another document in an `iframe`.